src/oscar/core/application.py

```diff
@@ -93,6 +96,9 @@ def post_process_urls(self, urlpatterns):
                 if decorator:
                     pattern.callback = decorator(pattern.callback)
 
+                if self.is_basket_enabled and pattern.name not in self.views_with_disabled_basket:
+                    pattern.callback.is_basket_enabled = True
```

Comment:
Should we add `self.views_with_enabled_basket`? This can be useful in case we want to add the basket to some views where config has `is_basket_enabled = False`.
E.g.
```python
if any([
    pattern.name in self.views_with_enabled_basket,
    self.is_basket_enabled and pattern.name not in self.views_with_disabled_basket,
]):
    pattern.callback.is_basket_enabled = True
```

Reply:
Looks like we do not need `self.views_with_enabled_basket` because we can use the `enable_basket` views decorator. With it, additionally, we will be able to enable the basket for views in apps that do not use `OscarConfig`.

Reply:
Yeah, I agree with your last conclusion.

Reply:
Thus, let's remove `views_with_disabled_basket` as well.

Reply:
> Thus, let's remove `views_with_disabled_basket` as well.

How in this case will be better to disable basket for views for apps where config has `is_basket_enabled = True`?

Reply:
We can use next approach - if basket is enabled on the application basis and disabled on the view basis just override application settings by the view. Which would finally disable basket for this view.

Reply:
Not sure how we can do this on the view level. Could you please explain a bit more detailed?

Reply:
On the view level we'll just add decorator and it will supersede application config.